.github/workflows/ci.yaml

```diff
@@ -43,10 +53,28 @@ jobs:
             parser-typechecker/.stack-work
             unison-core/.stack-work
             yaks/easytest/.stack-work
-          # Main hash key: the branch + commit hash. This should always result in a cache miss...
-          key: stack-work-0_${{matrix.os}}-${{github.ref}}-${{github.sha}}
-          # ...but then fall back on the latest cache stored on that branch.
-          restore-keys: stack-work-0_${{matrix.os}}-${{github.ref}}-
+            # Main cache key: commit hash. This should always result in a cache miss...
+          key: stack-work-0_${{matrix.os}}-${{github.sha}}
```

Comment:
I changed the caching key here because I learned from the docs of the `cache` action that it's already branch-based, so there's no reason to include `${{github.ref}}`.